src/DotNetWorker/Definition/DefaultFunctionDefinitionFactory.cs

```diff
@@ -18,11 +19,13 @@ internal class DefaultFunctionDefinitionFactory : IFunctionDefinitionFactory
         private static readonly Regex _entryPointRegex = new Regex("^(?<typename>.*)\\.(?<methodname>\\S*)$");
         private readonly IFunctionInvokerFactory _functionInvokerFactory;
         private readonly IFunctionActivator _functionActivator;
+        private readonly IOutputBindingsInfoFactory _outputBindingsInfoFactory;
 
-        public DefaultFunctionDefinitionFactory(IFunctionInvokerFactory functionInvokerFactory, IFunctionActivator functionActivator)
+        public DefaultFunctionDefinitionFactory(IFunctionInvokerFactory functionInvokerFactory, IFunctionActivator functionActivator, IOutputBindingsInfoFactory outputBindingsInfoFactory)
```

Comment:
Did we talk about renaming this to `Provider` or something like a `Reader`?

Reply:
Yeah we had mentioned maybe to review these surfaces later and make it all internal for now. So I hadn't. But I will update this to `Provider`. 

I would like to talk more on that later though when we review. To me, it still feels more like a factory trying to produce a an object based on some context.
Maybe I am not super familiar with specific differences in a Provider model vs Factory. I thought new object instantiating like this is typically a factory.